int64 left_encoder
int64 right_encoder
---
int8 result
